src/search/pdbs/cegar.cc

```diff
@@ -0,0 +1,589 @@
+            if (goal == task_goal.get_pair()) {
+                is_goal = true;
+                break;
+            }
+        }
+        if (!is_goal) {
+            cerr << "given goal is not a goal of the task." << endl;
+            utils::exit_with(utils::ExitCode::SEARCH_INPUT_ERROR);
+        }
+    }
+#endif
+    if (verbosity >= utils::Verbosity::NORMAL) {
+        utils::g_log << "options of the CEGAR algorithm for computing a pattern collection: " << endl;
+        utils::g_log << "max pdb size: " << max_pdb_size << endl;
+        utils::g_log << "max collection size: " << max_collection_size << endl;
+        utils::g_log << "wildcard plans: " << wildcard_plans << endl;
+        utils::g_log << "Verbosity: ";
+        switch (verbosity) {
+        case utils::Verbosity::SILENT:
+            utils::g_log << "silent";
+            break;
+        case utils::Verbosity::NORMAL:
+            utils::g_log << "normal";
+            break;
+        case utils::Verbosity::VERBOSE:
+            utils::g_log << "verbose";
+            break;
+        case utils::Verbosity::DEBUG:
+            utils::g_log << "debug";
+            break;
+        }
+        utils::g_log << endl;
+        utils::g_log << "max time: " << max_time << endl;
+        utils::g_log << "goal variables: ";
+        for (const FactPair &goal : goals) {
+            utils::g_log << goal.var << ", ";
+        }
+        utils::g_log << endl;
+        utils::g_log << "blacklisted variables: ";
+        if (blacklisted_variables.empty()) {
+            utils::g_log << "none";
+        } else {
+            for (int var : blacklisted_variables) {
+                utils::g_log << var << ", ";
+            }
+        }
+        utils::g_log << endl;
+    }
+}
+
+void add_cegar_options_to_parser(options::OptionParser &parser) {
+    parser.add_option<int>(
+        "max_pdb_size",
+        "maximum number of states per pattern database (ignored for the "
+        "initial collection consisting of singleton patterns for each goal "
+        "variable)",
+        "2000000",
+        Bounds("1", "infinity"));
+    parser.add_option<int>(
+        "max_collection_size",
+        "maximum number of states in the pattern collection (ignored for the "
+        "initial collection consisting of singleton patterns for each goal "
+        "variable)",
+        "20000000",
+        Bounds("1", "infinity"));
+    parser.add_option<bool>(
+        "wildcard_plans",
+        "if true, compute wildcard plans which are sequences of sets of "
+        "operators that induce the same transition; otherwise compute regular "
+        "plans which are sequences of single operators",
+        "true");
+    parser.add_option<double>(
+        "max_time",
+        "maximum time in seconds for the CEGAR algorithm (ignored for"
+        "computing initial collection)",
+        "infinity",
+        Bounds("0.0", "infinity"));
+}
+
+vector<FactPair> get_goals_in_random_order(
```

Comment:
You are right. I added it there because it is used only by those generators based on CEGAR. I could move it to `utils.h`, even though the scope of that file currently is much more general. But it probably doesn't hurt to have a few more specific functions there.